Jenkinsfile

```diff
@@ -92,56 +92,136 @@ pipeline {
+        * or git identities and is not open for SSH
+        */
+        stage('GCP - Stage A - Compute Engine') {
           steps {
+            echo '-- Allocating Google Compute Engine'
             script {
+              dir('ci/authn-gcp') {
+                stash name: 'get_gce_tokens_script', includes: 'get_gce_tokens_to_files.sh'
+              }
               node('executor-v2-gcp-small') {
+                echo '-- Google Compute Engine allocated'
+                echo '-- Get Google Cloud project name from Google metadata server'
+                env.GCP_PROJECT = sh (
+                  script: 'curl -s -H "Metadata-Flavor: Google" \
+                  "http://metadata.google.internal/computeMetadata/v1/project/project-id"',
+                  returnStdout: true
+                ).trim()
+                echo "Google Cloud project name: $GCP_PROJECT"
+                unstash 'get_gce_tokens_script'
                 sh '''
-                  retrieve_token() {
-                    local token_format="$1"
-                    local audience="$2"
-
-                    curl \
-                      -s \
-                      -H 'Metadata-Flavor: Google' \
-                      "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=${token_format}&audience=${audience}"
-                  }
-
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/test-app")" > "gcp_token_valid"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-existing")" > "gcp_token_non_existing_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-rooted/test-app")" > "gcp_token_non_rooted_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/test-app")" > "gcp_token_user"
-                  echo "$(retrieve_token "full" "conjur/non-existing/host/test-app")" > "gcp_token_non_existing_account"
-                  echo "$(retrieve_token "full" "invalid_audience")" > "gcp_token_invalid_audience"
-                  echo "$(retrieve_token "standard" "conjur/cucumber/host/test-app")" > "gcp_token_standard_format"
+                ./get_gce_tokens_to_files.sh || exit 1
                 '''
+                stash name: 'authnGceTokens', includes: 'gcp_token_*', allowEmpty:false
+              }
+            }
+          }
+          post {
+           failure {
+            script {
+              env.GCP_ENV_ERROR = "true"
+            }
+           }
+           success {
+            script {
+              env.GCE_TOKENS_FETCHED = "true"
+            }
+            echo '-- Finished fetching GCE tokens.'
+           }
+          }
+        }
+        stage('GCP - Stage B - Google Function') {
+          environment {
+            GCP_FETCH_TOKEN_FUNCTION = "fetch_token_${BUILD_NUMBER}"
+            IDENTITY_TOKEN_FILE = 'identity-token'
+            GCP_OWNER_SERVICE_KEY_FILE = "sa-key-file.json"
+            GCP_ZONE="us-central1"
```

Comment:
is this something that should be held in conjur and retrieved via summon?

Reply:
Which one? the zone?

Reply:
yes i also think the zone should be in conjur

Reply:
why? is this sensitive? will it change in the future?

Reply:
I don't think so, and I can extract it from GCE same as I do with the project.

Reply:
no but it isn't the first time we store these vars in conjur

Reply:
I think the best approach is to take it out of the gce node that infra is providing same as I get the project name. It will work for both ci and dev